Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -375,6 +460,160 @@ class SwiftObjectServerTests: SwiftSyncTestCase {
         }
     }
 
+    func testClientResetDiscardLocal() {
+        // Set client reset callbacks
+        do {
+            let user = try logInUser(for: basicCredentials())
+            let collection = setupMongoCollection(user: user, collectionName: "SwiftPerson")
+
+            // Define the blocks that will be passed into the sync configuration initializer.
+            // Use counter instead of an expectation because `waitFor*` methods will prematurely fail an expectation
+            var beforeCalledCount = 0
+            let beforeClientResetBlock: (Realm) -> Void = { local in
+                let results = local.objects(SwiftPerson.self)
+                XCTAssertEqual(results.count, 2)
+                let paul = results.filter("firstName == 'Paul'")
+                let john = results.filter("firstName == 'John'")
+                XCTAssertNotNil(paul)
+                XCTAssertNotNil(john)
+                beforeCalledCount += 1
+            }
+            // Use counter instead of an expectation because `waitFor*` methods will prematurely fail an expectation
+            var afterCalledCount = 0
+            let afterClientResetBlock: (Realm, Realm) -> Void = { before, after in
+                // Expect the local realm that was overwritten to have had 2 objects before it was overwritten.
+                let results = before.objects(SwiftPerson.self)
+                XCTAssertEqual(results.count, 2)
+                let paul = results.filter("firstName == 'Paul'")
+                let john = results.filter("firstName == 'John'")
+                XCTAssertNotNil(paul)
+                XCTAssertNotNil(john)
+                // Expect the server realm that overwrote the local realm to have had 1 object before client reset.
+                let results2 = after.objects(SwiftPerson.self)
+                XCTAssertEqual(results2.count, 1)
+                let paul2 = results2.filter("firstName == 'Paul'")
+                let john2 = results2.filter("firstName == 'John'")
+                XCTAssertFalse(paul2.isEmpty)
+                XCTAssert(john2.isEmpty)
+                afterCalledCount += 1
+            }
+
+            var configuration = user.configuration(partitionValue: #function,
+                                                   clientResetMode: .discardLocal,
+                                                   notifyBeforeClientReset: beforeClientResetBlock,
+                                                   notifyAfterClientReset: afterClientResetBlock)
```

Comment:
Should this maybe be `clientResetMode: .discardLocal(beforeReset: beforeClientResetBlock, afterReset: afterClientResetBlock)`? It would eliminate one of the ways to call `configuration()` incorrectly.